content/slice/language-guide/exception.md

```diff
@@ -76,29 +76,39 @@ exception TranslationException {
 ```
 
 ```csharp
-public partial class TranslationException
-    : SliceException
+public partial class TranslationException :
+    SliceException
 {
-    public TranslationErrorCode ErrorCode;
-    public string? DetectedLanguage;
+    public TranslationErrorCode ErrorCode { get; set; }
+    public string? DetectedLanguage { get; set; }
+
+    // Parameterless constructor
+    public TranslationException()
+    {
+    }
 
     // Primary constructor
     public TranslationException(
         TranslationErrorCode errorCode,
-        string? detectedLanguage,
-        string? message = null,
-        System.Exception? innerException = null)
-        : base(message, innerException)
+        string? detectedLanguage)
+    {
+        this.ErrorCode = errorCode;
+        this.DetectedLanguage = detectedLanguage;
+    }
+
+    // Secondary constructor
+    public TranslationException(
+        TranslationErrorCode errorCode)
     {
-        ...
+        this.ErrorCode = errorCode;
     }
 }
 ```
 
 {% /aside %}
 
-The mapped C# class provides a primary constructor with parameters for all its fields, plus an optional message and an
-optional inner exception (like most exceptions in C#).
+The mapped C# class provides a parameterless constructor and a primary constructor with parameters for all its fields.
+It also provides a secondary constructor if any Slice field has an optional type.
```

Comment:
I think it would be nice to refactor this sentence to mention that this secondary constructor only features the
non-optional fields. Just saying the secondary constructor exists is not very useful otherwise.

Reply:
Fixed.